tests/qt/qmetamethod/tst_qmetamethod.cpp

```diff
@@ -39,16 +39,22 @@ class tst_QMetaMethod : public QObject
     W_OBJECT(tst_QMetaMethod)
 
 private slots:
-    void method_data(); W_SLOT(method_data, W_Access::Private)
-    void method(); W_SLOT(method, W_Access::Private)
+    void method_data();
```

Comment:
Why is that required?  
I think it is better to have the W_SLOT next to the method, this way we can easily see that the W_SLOT macro was not forgotten.

Reply:
It's not absolutely required. If QtCreator finds one slot, it will run all tests.

But QtCreator will not be able to find the other slots that way. Therefore you cannot limit test runs to these tests.

Reply:
I am not using QtCreator, so i'm not sure what you mean by QtCreator "finding" the test.
Why should QtCreator find the test? Is it not just running the tests by compiling and running the test program?

Reply:
> I am not using QtCreator

That's strange. There are tests that were structured like that. - I made it consistent.

> Is it not just running the tests by compiling and running the test program?

Most build systems support running tests, but QtCreator does not support this.
You can select one executable and run it or use the test runner.
The test runner only runs executables with tests, that QtCreator had discovered before. 

> Why should QtCreator find the test?

See https://doc.qt.io/qtcreator/creator-autotest.html
Unfortunately QtCreator does not ask the build system but scans the source files using the C++ Code Model.
See: https://github.com/qt-creator/qt-creator/blob/master/src/plugins/autotest/qtest/qttestparser.cpp
If a project contains at least one test - it is shown and run by the test runner.
If all tests are discovered it's possible to select the tests to run.

Maybe it's not the best idea to use Verdigris to write the tests. If something in Verdigris is wrong, it's possible that unit tests pass because not tests were run.